lib/ecto/adapters/sql.ex

```diff
@@ -141,11 +146,17 @@ defmodule Ecto.Adapters.SQL do
   def to_sql(kind, repo, queryable) do
     adapter = repo.__adapter__
 
-    {_meta, prepared, params} =
+    plan =
       Ecto.Queryable.to_query(queryable)
       |> Ecto.Query.Planner.query(kind, repo, adapter)
-
-    {prepared, params}
+    case plan do
+      {:execute, _meta, %{statement: statement}, params} ->
```

Comment:
Curiosity, why do we have the statement sometimes? Is this when it comes after prepare_execute? I am afraid this is being specific to PG. Can't we call DBCOnnection to string or something? Or can we get rid of the difference in the first place?